### PR TITLE
ui: create connected components for database pages

### LINKED ISSUE #97885: use cluster-ui database component on CC Console
Currently we're using a separate component on CC Console. This should be replaced by the cluster-ui Database/Tables component.

Jira issue: CRDB-24945

Epic: CRDB-25476

### PR DESCRIPTION
**Pull request by THardy98**
Resolves: https://github.com/cockroachdb/cockroach/issues/97885

Create connected components for the database pages

### CODE REVIEW

File: pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+          nodes,
+          nodeRegionsByIDSelector(state),
+          selectIsTenant(state),
+        ),
+      },
+    };
+  });
+};
+
+export const mapStateToProps = createSelector(
```
Comment (xinhaoz) [inline pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts:100]: Is there a reason why this is created as a selector? I believe we talked about why this is unnecessary for the other cases.

File: pkg/ui/workspaces/cluster-ui/src/databaseTablePage/databaseTablePageConnected.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        totalReads: longToInt(indexStat.statistics?.stats?.total_read_count),
+        lastUsed: lastUsed,
+        lastUsedType: lastUsedType,
+        indexRecommendations,
+      };
+    },
+  );
+}
+
+export const mapStateToProps = createSelector(
```
Comment (xinhaoz) [inline pkg/ui/workspaces/cluster-ui/src/databaseTablePage/databaseTablePageConnected.ts:92]: Same thing here.

File: pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+import {
+  combineLoadingErrors,
+  generateTableID,
+  getNodesByRegionString,
+  normalizePrivileges,
+  normalizeRoles
+} from "../util";
+import {selectTableDetails} from "../store/databaseTableDetails/tableDetails.selector";
+
+const selectTables = (state: AppState, props: RouteComponentProps): DatabaseDetailsPageDataTable[] => {
```
Comment (xinhaoz) [inline pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts:47]: Is all this selector logic now being duplicated across db-console and cluster-ui?

Comment (THardy98) [reply pkg/ui/workspaces/cluster-ui/src/databaseDetailsPage/databaseDetailsConnected.ts]: Refactored to use common selector logic between database components in db-console & cluster-ui. See `databasesCommon.selectors.ts`

**Review by xinhaoz — COMMENTED** [review]
There seems to be a bit more than just 'creating connected db pages' here. Mind updating with all significant changes in the commit message(s)?

This is a very large PR, so I'd be happy to set up some time to go over it in a call / go over how to split this up into smaller commits.

**Comment by THardy98** [comment]
Closing this PR in favour of multiple smaller PRs. I've addressed the suggested comments here on said smaller PRs.
Starting with: https://github.com/cockroachdb/cockroach/pull/103979

**Comment by THardy98** [comment]
DB Details: https://github.com/cockroachdb/cockroach/pull/104168

**Pull request closed without merging**